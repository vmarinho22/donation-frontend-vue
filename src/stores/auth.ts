import { defineStore } from 'pinia'
import { ref } from 'vue'
import api from '@/utils/api'
import type { SignInResponse } from '@/types/auth'

type Login = {
  email: string
  password: string
}

export const useAuth = defineStore('auth', () => {
  const loginAuth = ref<Login>({
    email: '',
    password: ''
  })

  async function signIn(): Promise<string> {
    const { data } = await api.post<SignInResponse>('/auth/signin', loginAuth.value)

    localStorage.setItem('access_token', data.token)

    return data.userId
  }

  return { loginAuth, signIn }

})
