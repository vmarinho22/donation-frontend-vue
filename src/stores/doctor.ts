import { defineStore } from 'pinia'
import { ref } from 'vue'
import api from '@/utils/api'
import type { Doctor } from '@/types/doctors'

export const initialDoctorData: Doctor = {
  id: '',
  firstName: '',
  lastName: '',
  email: '',
  socialName: '',
  specialty: '',
  registrationNumber: '',
  subspecialties: null,
  emergencyTelContact: '',
  disabled: false,
  disabledAt: null,
  userId: '',
  createdAt: new Date(),
  updatedAt: new Date()
}

export const useDoctor = defineStore(
  'doctor',
  () => {
    const doctor = ref<Doctor>(initialDoctorData)

    async function getDoctorByUserId(userId: string) {
      const response = await api.get<Doctor>(`/doctors/by-user/${userId}`)
      return response.data
    }

    async function fetchAndUpdateDoctor(userId: string) {
      const response = await getDoctorByUserId(userId)
      doctor.value = response
    }

    async function clearDoctorData() {
      doctor.value = initialDoctorData
      localStorage.removeItem('doctor')
    }

    function updateDoctor(doctorPartialData: Partial<Doctor>) {
      doctor.value = Object.assign(doctor.value, doctorPartialData)
    }

    return {
      doctor,
      fetchAndUpdateDoctor,
      getDoctorByUserId,
      clearDoctorData,
      updateDoctor
    }
  },
  {
    persist: true
  }
)
