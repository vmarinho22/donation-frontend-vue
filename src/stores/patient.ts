import { defineStore } from 'pinia'
import type { Patient } from '@/types/patient'
import { ref } from 'vue'
import type { MedicalRecord } from '@/types/medicalRecord'
import api from '@/utils/api'

type RegisterPatientState = {
  birthDate: Date
  sex: 'man' | 'woman' | 'other'
  gender: string | null
  age: number
  weight: number
  height: number
  rg: string
  bloodId: string
}

export const initialPatientRegisterState: RegisterPatientState = {
  birthDate: new Date(),
  sex: 'man',
  gender: null,
  age: 0,
  weight: 0,
  height: 0,
  rg: '',
  bloodId: ''
}

export const initialPatientState: Patient = {
  id: '',
  ...initialPatientRegisterState,
  userId: '',
  medicalRecordId: '',
  createdAt: new Date(),
  updatedAt: null
}

const initialMedicalRecordRegisterState = {
  hasChronicDiseases: false,
  chronicDiseases: undefined,
  hasMedicalConditions: false,
  medicalConditions: undefined,
  hasPreviousSurgeries: false,
  previousSurgeries: undefined,
  hasBloodBorneDiseases: false,
  hasCommunicableDiseases: false,
  communicableDiseases: undefined,
  hasIst: false,
  ist: undefined,
  useIllicitDrugs: false,
  useInjectingDrugs: false,
  hasPracticeUnprotectedSex: false,
  hadPregnancy: false,
  recentlyBreastfed: false,
  lastBreastfeeding: undefined,
  notes: undefined
}

export const initialMedicalRecordState: MedicalRecord = {
  id: '',
  ...initialMedicalRecordRegisterState,
  createdAt: new Date(),
  updatedAt: null
}

export const usePatient = defineStore(
  'patient',
  () => {
    const patient = ref<Patient>(initialPatientState)
    const registerPatientState = ref<RegisterPatientState>(initialPatientRegisterState)
    const medicalRecord = ref<MedicalRecord>(initialMedicalRecordState)
    const registerMedicalRecord = ref(initialMedicalRecordRegisterState)

    function updatePatient(userPartialData: Partial<Patient>) {
      patient.value = Object.assign(patient.value, userPartialData)
    }

    function updateRegisterState(registerStatePartialData: Partial<RegisterPatientState>) {
      registerPatientState.value = Object.assign(
        registerPatientState.value,
        registerStatePartialData
      )
    }

    function updateMedicalRecord(
      medicalRecordPartialData: Partial<Omit<MedicalRecord, 'id' | 'createdAt' | 'updatedAt'>>
    ) {
      medicalRecord.value = Object.assign(medicalRecord.value, medicalRecordPartialData)
    }

    async function fetchPatientDataByUserId(userId: string) {
      const patientData = await api.get<Patient>(`/patients/by-user/${userId}`).then(res => res.data)

      updatePatient(patientData)
    }

    async function fetchMedicalRecord(id: string) {
      const medicalRecordData = await api
        .get<MedicalRecord>(`/medical-records/${id}`)
        .then(res => res.data)

      updateMedicalRecord(medicalRecordData)
    }

    return {
      patient,
      registerState: registerPatientState,
      medicalRecord,
      registerMedicalRecord,
      updatePatient,
      updateRegisterState,
      updateMedicalRecord,
      fetchPatientDataByUserId,
      fetchMedicalRecord
    }
  },
  {
    persist: true
  }
)
