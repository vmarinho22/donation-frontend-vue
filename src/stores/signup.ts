import { ref } from 'vue'
import { defineStore } from 'pinia'
import type { Role } from '@/types/roles'

export type UserSignUp = {
  email: string
  password: string
  confirmPassword: string
  cpf: string
  role: Role
}

export type ProfileSignUp = {
  firstName: string
  lastName: string
  socialName: string | null
  photoUrl: string
  phone: string
}

export type AddressSignUp = {
  postalCode: string
  street: string
  number: number | null
  hasNumber: boolean
  complement: string | null
  district: string
  city: string
  state: string
  country: string
  uf: string
}

export type SignUpState = {
  user: UserSignUp
  profile: ProfileSignUp
  address: AddressSignUp
}

const initialUserData = {
  email: '',
  password: '',
  confirmPassword: '',
  cpf: '',
  role: 'patient'
} as UserSignUp

const initialProfileData = {
  firstName: '',
  lastName: '',
  socialName: null,
  photoUrl:
    'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png',
  phone: ''
}

const initialAddressData = {
  postalCode: '',
  street: '',
  hasNumber: false,
  number: null,
  complement: null,
  district: '',
  city: '',
  state: '',
  country: '',
  uf: ''
}

export const useSignUp = defineStore('signup', () => {
  const user = ref<UserSignUp>(initialUserData)

  const profile = ref<ProfileSignUp>(initialProfileData)

  const address = ref<AddressSignUp>(initialAddressData)

  function updateUser(key: keyof UserSignUp, value: any) {
    user.value[key] = value
  }

  function updateProfile(key: keyof ProfileSignUp, value: any) {
    profile.value[key] = value
  }

  function updateAddress(key: keyof AddressSignUp, value: any) {
    address.value[key] = value as never
  }

  function clearAllData() {
    user.value = Object.assign(user.value, initialUserData)
    profile.value = Object.assign(profile.value, initialProfileData)
    address.value = Object.assign(address.value, initialAddressData)
  }

  return { user, profile, address, updateUser, updateProfile, updateAddress, clearAllData }
})
