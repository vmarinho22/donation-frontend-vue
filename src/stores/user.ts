import { defineStore } from 'pinia'
import type { User } from '@/types/user'
import { ref } from 'vue'

export const initialState: User = {
  id: '',
  firstName: '',
  lastName: '',
  socialName: null,
  email: '',
  cpf: '',
  lang: 'pt_br',
  photoUrl: '',
  role: 'patient',
  phone: '',
  address: {
    postalCode: '',
    street: '',
    number: '',
    hasNumber: false,
    district: '',
    city: '',
    state: '',
    country: '',
    uf: ''
  }
}


export const useUser = defineStore('user', () => {
  const user = ref<User>(initialState)

  function updateUser(userPartialData: Partial<User>) {
    user.value = Object.assign(user.value, userPartialData)
  }

  return { user, updateUser }
}, {
  persist: true
})
