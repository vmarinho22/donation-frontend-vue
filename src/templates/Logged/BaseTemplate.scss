.base-template {
  position: relative;
  .nav-section {
    width: 20%;
  }

  .content-section {
    width: 80%;
  }

  .role {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    margin-bottom: 16px;
    gap: 16px;
  }

  .user-info {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    gap: 16px;
    margin-top: 16px;
    margin-bottom: 16px;
  }

  .config-buttons {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    align-content: center;
    justify-content: center;
  }
}