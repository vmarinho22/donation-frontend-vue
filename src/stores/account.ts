import { defineStore } from 'pinia'
import { useUser, initialState } from "./user";
import { computed } from 'vue';

export const useAccount = defineStore('account', () => {
  const { user, updateUser } = useUser()

  const account = computed(() => {
    return {
      user
    }
  })

  function signOff() {
    updateUser(initialState)
    localStorage.removeItem('user')
    localStorage.removeItem('patient')
  }

  return { account, signOff }
})