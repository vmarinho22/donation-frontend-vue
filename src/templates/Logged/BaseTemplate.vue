<template>
  <v-layout class="base-template">
    <v-navigation-drawer v-model="drawer" :rail="rail" permanent @click="rail = false">
      <v-list>
        <v-list-item
          :prepend-avatar="account.user.photoUrl"
          :title="
            account.user.socialName
              ? account.user.socialName
              : `${account.user.firstName.split(' ')[0]} ${account.user.lastName}`
          "
          :subtitle="$t(`roles.${account.user.role}`)"
        >
          <template v-slot:append>
            <v-btn variant="text" icon="mdi-chevron-left" @click.stop="rail = !rail" />
          </template>
        </v-list-item>
      </v-list>

      <v-list density="compact" nav v-for="item in menu" v-bind:key="item.id">
        <v-list-item
          :prepend-icon="item.icon"
          :title="$t(`menu.${item.localeKey}`)"
          :value="item.id"
          :active="routeInfo.fullPath.startsWith(`/${item.id}`)"
          @click="goTo(`/${item.id}`)"
        />
      </v-list>

      <template v-slot:append>
        <div class="config-buttons">
          <toggle-theme />
          <v-menu transition="slide-y-transition">
            <template v-slot:activator="{ props }">
              <v-btn icon="mdi-account-circle" variant="text" v-bind="props"></v-btn>
            </template>
            <v-list>
              <v-list-item @click="console.log('config')">
                <v-list-item-title>{{ $t('menu.userMenu.config') }}</v-list-item-title>
              </v-list-item>
              <v-list-item @click="signOut()">
                <v-list-item-title>{{ $t('menu.userMenu.signOut') }}</v-list-item-title>
              </v-list-item>
            </v-list>
          </v-menu>
          <br />
        </div>
      </template>
    </v-navigation-drawer>

    <v-main>
      <v-toolbar
        density="compact"
        title="Donation System"
        color="transparent"
        style="margin-bottom: 16px"
      >
        <toggle-theme />
      </v-toolbar>
      <v-container fluid>
        <slot />
      </v-container>
    </v-main>
  </v-layout>
</template>

<script setup lang="ts">
import { ref } from 'vue'
import ToggleTheme from '@/components/Buttons/ToggleTheme/ToggleTheme.vue'
import { useAccount } from '@/stores/account'
import type { Menu } from '@/types/menu'
import { useRoute, useRouter } from 'vue-router'

const { menu } = defineProps<{
  menu: Menu[]
}>()

const drawer = ref(true)
const rail = ref(true)

const { account, signOff } = useAccount()
const router = useRouter()
const routeInfo = useRoute()

function goTo(route: string) {
  router.push(route)
}

function signOut() {
  signOff()
  router.push('/auth/login')
}
</script>

<style scoped lang="scss" src="./BaseTemplate.scss" />
