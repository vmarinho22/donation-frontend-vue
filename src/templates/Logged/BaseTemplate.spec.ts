import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import BaseTemplate from './BaseTemplate.vue'
import vuetify from '@/tests/plugins/vuetify'
import { useAccount } from '@/stores/account'

import { createTestingPinia } from '@pinia/testing'
import mockedUser from '@/tests/mocks/user'

describe('BaseTemplate', () => {
 

  it('renders the component', () => {
    const wrapper = mount(BaseTemplate, {
      global: {
        plugins: [vuetify, createTestingPinia()]
      },
      props: {
        menu: [
          {
            id: '1',
            localeKey: 'Home',
            icon: 'mdi-home',
          }
        ]
      }
    })

    const store = useAccount()

    store.account.user = mockedUser
    expect(wrapper.html()).toMatchSnapshot();
  })
})