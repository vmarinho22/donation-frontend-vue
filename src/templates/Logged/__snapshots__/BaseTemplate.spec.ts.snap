// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseTemplate > renders the component 1`] = `
"<div data-v-980a0d65=\\"\\" class=\\"v-layout base-template\\">
  <nav class=\\"v-navigation-drawer v-navigation-drawer--left v-navigation-drawer--rail v-navigation-drawer--active v-theme--light\\" style=\\"left: 0px; z-index: 1004; transform: translateX(0%); position: fixed; transition: none;\\" data-v-980a0d65=\\"\\">
    <!---->
    <!---->
    <div class=\\"v-navigation-drawer__content\\">
      <div data-v-980a0d65=\\"\\" class=\\"v-list v-theme--light bg-transparent v-list--density-default v-list--one-line\\" tabindex=\\"0\\" role=\\"listbox\\">
        <div data-v-980a0d65=\\"\\" class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
          <!----><span class=\\"v-list-item__underlay\\"></span>
          <!---->
          <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
            <div class=\\"v-list-item-title\\"> </div>
            <div class=\\"v-list-item-subtitle\\">roles.patient</div>
            <!---->
          </div>
          <div class=\\"v-list-item__append\\"><button data-v-980a0d65=\\"\\" type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
              <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-chevron-left mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
              <!---->
              <!---->
            </button>
            <div class=\\"v-list-item__spacer\\"></div>
          </div>
        </div>
      </div>
      <div data-v-980a0d65=\\"\\" class=\\"v-list v-list--nav v-theme--light bg-transparent v-list--density-compact v-list--one-line\\" tabindex=\\"0\\" role=\\"listbox\\">
        <div data-v-980a0d65=\\"\\" class=\\"v-list-item v-list-item--link v-list-item--nav v-theme--light v-list-item--density-compact v-list-item--one-line v-list-item--rounded v-list-item--variant-text\\" tabindex=\\"-2\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
          <div class=\\"v-list-item__prepend\\">
            <!----><i class=\\"mdi-home mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\" density=\\"compact\\"></i>
            <div class=\\"v-list-item__spacer\\"></div>
          </div>
          <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
            <div class=\\"v-list-item-title\\">menu.Home</div>
            <!---->
            <!---->
          </div>
          <!---->
        </div>
      </div>
    </div>
    <div class=\\"v-navigation-drawer__append\\">
      <div data-v-980a0d65=\\"\\" class=\\"config-buttons\\"><button data-v-980a0d65=\\"\\" type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
          <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-weather-night mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
          <!---->
          <!---->
        </button><button data-v-980a0d65=\\"\\" type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-5\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
          <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-account-circle mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
          <!---->
          <!---->
        </button>
        <!----><br data-v-980a0d65=\\"\\">
      </div>
    </div>
  </nav>
  <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
    <!---->
  </transition-stub>
  <main data-v-980a0d65=\\"\\" class=\\"v-main\\" style=\\"--v-layout-left: 56px; --v-layout-right: 0px; --v-layout-top: 0px; --v-layout-bottom: 0px; transition: none !important; min-height: 100vh;\\">
    <header data-v-980a0d65=\\"\\" class=\\"v-toolbar v-toolbar--density-compact bg-transparent v-theme--light v-locale--is-ltr\\" style=\\"margin-bottom: 16px;\\">
      <!---->
      <div class=\\"v-toolbar__content\\" style=\\"height: 48px;\\">
        <!---->
        <div class=\\"v-toolbar-title\\">
          <div class=\\"v-toolbar-title__placeholder\\">Donation System
            <!---->
          </div>
        </div><button data-v-980a0d65=\\"\\" type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
          <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-weather-night mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
          <!---->
          <!---->
        </button>
        <!---->
      </div>
      <transition-stub name=\\"expand-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
        <!---->
      </transition-stub>
    </header>
    <div data-v-980a0d65=\\"\\" class=\\"v-container v-container--fluid v-locale--is-ltr\\"></div>
  </main>
</div>"
`;
